cast/src/lib.rs

```diff
@@ -852,6 +853,38 @@ impl SimpleCast {
         let calldata = encode_args(&func, args)?;
         Ok(format!("0x{}", calldata.to_hex::<String>()))
     }
+
+    /// Fetches source code of verified contracts from etherscan.
+    ///
+    /// ```
+    /// # use cast::SimpleCast as Cast;
+    /// # use ethers_core::types::Chain;
+    /// 
+    /// # async fn foo() -> eyre::Result<()> {
+    ///     assert_eq!(
+    ///             "/*
+    ///             - Bytecode Verification performed was compared on second iteration -
+    ///             This file is part of the DAO.....",
+    ///         Cast::etherscan_source(Chain::Mainnet, "0xBB9bc244D798123fDe783fCc1C72d3Bb8C189413".to_string(), "<etherscan_api_key>".to_string()).await.unwrap().as_str()
+    ///     );
+    /// #    Ok(())
+    /// # }
+    /// ```
+    pub async fn etherscan_source(
+        chain: Chain,
+        contract_address: String,
+        etherscan_api_key: String,
+    ) -> Result<String> {
+        let client = Client::new(chain, etherscan_api_key).unwrap();
```

Comment:
```suggestion
        let client = Client::new(chain, etherscan_api_key)?;
```

Reply:
Yeah cool. I was under the impression that they both are the same thing but just realized that `unwrap()` panics, so makes sense.